src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,148 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2022 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+#include <ripple/app/misc/AMM.h>
+#include <ripple/json/json_value.h>
+#include <ripple/ledger/ReadView.h>
+#include <ripple/net/RPCErr.h>
+#include <ripple/protocol/Issue.h>
+#include <ripple/rpc/Context.h>
+#include <ripple/rpc/impl/GRPCHelpers.h>
+#include <ripple/rpc/impl/RPCHelpers.h>
+#include <grpcpp/support/status.h>
+
+namespace ripple {
+
+std::optional<AccountID>
+getAccount(Json::Value const& v, Json::Value& result)
+{
+    std::string strIdent(v.asString());
+    AccountID accountID;
+
+    if (auto jv = RPC::accountFromString(accountID, strIdent))
+    {
+        for (auto it = jv.begin(); it != jv.end(); ++it)
+            result[it.memberName()] = (*it);
+
+        return std::nullopt;
+    }
+    return std::optional<AccountID>(accountID);
```

Comment:
This returns a seated accountID that's empty in some cases. That doesn't seem right. Also, just return `accountID`, why is this explicitly wrapped in an optional?